### ISSUE TITLE
Interpretation for predict with truncated_poisson family

### ISSUE THREAD

**Issue by pitakakariki** [glmmTMB/glmmTMB#634]
I'm having trouble working out the correct interpretation for the predicted values from a hurdle model. I thought based on the documentation (which I may have misread) that `type="conditional"` would give me the means of the truncated Poisson part, but I was getting predictions that were less than one, so it seems to be giving me the untruncated Poisson means. But this means that the `type="response"` means are probably not what is expected either.

```
set.seed(42)
N <- 100
df <- data.frame(y=rpois(N, 1) * rbinom(N, 1, 0.5), x=rnorm(N))

fm1 <- glmmTMB(y ~ x, zi= ~ 1, family=truncated_poisson(link="log"), data=df)
```

To save a bit of typing:

```
zp1 <- predict(fm1, type="zprob")
cm1 <- predict(fm1, type="conditional")
mu1 <- predict(fm1, type="response")
```

The `zprob` predictions are fine:

```
> mean(df$y > 0)
[1] 0.42
> mean((1-zp1))
[1] 0.4200004
```

The `conditional` means are fine, as long as they're interpreted as the means of the pre-truncated Poisson:

```
> mean(df$y[df$y > 0])
[1] 1.690476
> mean(cm1 / (1 - exp(-cm1)))
[1] 1.68947
```

But the `response` means are a bit strange - looks like they're just using the default `mu*(1-p)` which works for zero-inflated but not for hurdle models?

```
> mean(df$y)
[1] 0.71
> mean(mu1)
[1] 0.4863494
> mean((1-zp1) * cm1 / (1-exp(-cm1)))
[1] 0.7095783
> 
> all.equal(cm1 * (1-zp1), mu1)
[1] TRUE
```

**Comment by mebrooks** [comment]
I guess that we need to make a new value of `ziPredictCode` for hurdle models. Then the harder part is that the formula is different depending on the distribution of the conditional model, e.g. negative binomial models would have a different denominator than in the Poisson example above. The relevant families are `truncated_poisson_family`, `truncated_genpois_family`, `truncated_compois_family`, `truncated_nbinom1_family`, `truncated_nbinom2_family`. Once we figure out the best way to handle that, part of it can go in this [switch statement](https://github.com/glmmTMB/glmmTMB/blob/ddfa5102104f08d0ee8743ca38849d80c2e66da8/glmmTMB/src/glmmTMB.cpp#L888).

**Comment by joeschofield0** [comment]
Hi.  I am experiencing this issue with a zero-truncated Poisson regression, where some predicted values are < 1.  I appreciate this has been flagged as a bug and is still open.  In the meantime, please could you advise if there is any workaround I can apply to fix this in my current analysis?

**Comment by bbolker** [comment]
I have something that fixes this (proximally), but I'm worried about the number of factorial test cases (fitting non-truncated vs truncated distributions, with/without zero-inflation, conditional vs response predictions): https://github.com/glmmTMB/glmmTMB/blob/truncated_predict_fix/misc/trunc_predict.R .

My test code works for truncated Poisson and NB1 but *not* for trunc NB2 ... not sure why ...

**Comment by bbolker** [comment]
More checking of the `trunc_predict.R` file linked above suggests that things are OK: the mean of the non-zero values is a bit off of the mean of the conditional responses, but I'm not quite sure why that's true in the first place.

- need to add more tests
- need to emphasize in docs/NEWS file that the conditional prediction for truncated distributions now takes the truncation into account

(Pretty close to making a pull request ...)

**Comment by bbolker** [comment]
should be fixed by 1e37cbecdbe607 (feel free to reopen if there are further issues)
